src/hardware/input/intel8042.cpp

```diff
@@ -992,6 +1010,7 @@ static uint8_t read_data_port(io_port_t, io_width_t) // port 0x60
 	}
 
 	if (is_data_from_kbd) {
+		PIC_DeActivateIRQ(get_irq_keyboard());
 		assert(waiting_bytes_from_kbd);
```

Comment:
This seems like a pretty big change, is this the confirmed real hardware behaviour? Could you test on real hardware, perhaps? I can run some simple test program on hardware if you give me a binary.

A lot of the games (I'd say most) read from port 60h because it's a lot faster than calling the BIOS interrupts. Demos too, I think I always just read from port 60h in my own programs.

I'm just wondering if this change fixes _this_ particular game but deviates from actual hardware behaviour, we'de get issues in some other games that read from port 60h and expect some specific keyboard IRQ behaviour as well.

Reply:
I have looked at QEMU source code and it looked to me like it behaves the same way. Definitely DOSBox-X does something similar, on purpose, to fix some issues:

https://github.com/joncampbell123/dosbox-x/blob/230b1e3d93d1c79039404aa29bc1b9de4361e945/src/hardware/keyboard.cpp#L374

I haven't tested the legacy behavior, as the i8042 emulation there is, IMHO, quite messy.

I don't even know how can I check the `PIC_ActivateIRQ` / `PIC_DeActivateIRQ` status on the real hardware - maybe it's possible to check this by reading some PIC status, I'll try to investigate.

Reply:
Thanks @FeralChild64, I appreciate the info and you having a further look into it. I'm just wary of making fundamental changes to areas that affect practically all programs, given my recent experiences with refactoring the VGA code 😅 

Given it's practically impossible for us to test thousands of program looking for keyboard regressions, I think it would a better use of our time to confirm we're replicating the exact hardware behaviour, ideally with the simplest possible test program on hardware.